File: LeetCode-Cpp/SearchinRotatedSortedArray.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size() - 1;

        while (left <= right) {
            int mid = left + (right - left) / 2;

            if (nums[mid] == target) {
                return mid; // hedef bulundu
            }

            // Sol taraf sıralı mı?
            if (nums[left] <= nums[mid]) {
                // Target sol tarafta mı?
                if (nums[left] <= target && target < nums[mid]) {
                    right = mid - 1; // sola kay
                } else {
                    left = mid + 1; // sağa kay
                }
            }
            // Yoksa sağ taraf sıralı
            else {
                // Target sağ tarafta mı?
                if (nums[mid] < target && target <= nums[right]) {
                    left = mid + 1; // sağa kay
                } else {
                    right = mid - 1; // sola kay
                }
            }
        }
        return -1; // bulunamazsa
    }
};

File: LeetCode-Python/SearchinRotatedSortedArray.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left, right = 0, len(nums) - 1

        while left <= right:
            mid = (left + right) // 2

            if nums[mid] == target:
                return mid  # hedef bulundu

            # Sol taraf sıralı mı?
            if nums[left] <= nums[mid]:
                if nums[left] <= target < nums[mid]:
                    right = mid - 1  # sola kay
                else:
                    left = mid + 1   # sağa kay
            # Sağ taraf sıralı
            else:
                if nums[mid] < target <= nums[right]:
                    left = mid + 1   # sağa kay
                else:
                    right = mid - 1  # sola kay

        return -1  # bulunamadı
